src/app/app-pipes/filter-by.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Course } from '@app/app-models';
+
+/** Filters out courses whose title matches the given string. */
+@Pipe({
+  name: 'filterBy'
+})
+export class FilterByPipe implements PipeTransform {
+  transform(courses: Course[], ...args: string[]): Course[] {
```

Comment:
Here we could try:

```typescript
    transform(courses: Course[], serachTerm: string, ...args: string[]): Course[] {
```

Reply:
Got it - I'll refactor this in next task